docker/build-tools/Dockerfile

```diff
@@ -689,6 +798,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 
 COPY --from=binary_tools_context /out/ /
 COPY --from=binary_tools_context /usr/local/go /usr/local/go
+COPY --from=gn_context /gn/gn /usr/local/bin/gn
```

Comment:
How come we need `gn` when it doesn't seem like we had it before?

Reply:
gn is wee8 required for arm64

Reply:
gn is required. It was probably proxy that download it. Not hurt if we move it here.

Reply:
Chat with @howardjohn We should not use master. For now we can use the commit 5ed3c9cc67b090d5e311e4bd2aba072173e82db9 as in the gn in centos build

Reply:
And maybe add a comment to keep it in sync with Dockerfile.centos

Reply:
@lambdai because `master` may compile failed ( i don't know why 🤣 , the `5ed3c9cc67b090d5e311e4bd2aba072173e82db9` is from `envoy-build-tools`).

@howardjohn 
we could refactor to split common install scripts to shell files to support ubuntu and centos both, like envoy-build-tools did https://github.com/envoyproxy/envoy-build-tools/tree/main/build_container

Reply:
+1, would be great to have a shared location